### PR TITLE
Refactor HighlightUsageTagger

### PR DESCRIPTION
**Pull request by smoothdeveloper**
So that logic is separated from FSharpVSPowerTools.Logic project with a new HighlightUsageInFile module in FSharpVSPowerTools.Core project

### CODE REVIEW

File: src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        |> Seq.groupBy (fst >> Path.GetFullPathSafe)
+        |> Seq.collect (fun (_, symbolUses) -> 
+            symbolUses 
+            |> Seq.map snd 
+            |> Seq.distinctBy (fun s -> s.RangeAlternate))
+        |> Seq.toArray
+
+type FileName = string
+type GetCheckResults = FileName -> Async<ParseAndCheckResults option>
+type FCSZeroBasedRange = int * int * int * int
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs]: Remove `FCS` prefix. It doesn't add more information.

File: src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+type CurrentLine = {Line: string; File: FileName; Range: FCSZeroBasedRange}
+    with
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs]: The use of `with` is weird.
You can use indentation:

```fsharp
type CurrentLine = 
    {Line: string; File: FileName; Range: FCSZeroBasedRange}
    member x.EndLine = ...
```

File: src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        member x.EndLine =
+            let _, _, endLine, _ = x.Range
+            endLine
+
+module HighlightUsageInFile =
+    open FSharpVSPowerTools
+    open Microsoft.FSharp.Compiler.SourceCodeServices
+    [<NoComparisonAttribute(* due to FSharpSymbol*) >]
+    type HighlightUsageInFileResult =
+    | UsageInFile of FSharpSymbol * string * FSharpSymbolUse array
```
Comment (dungpa) [inline src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs]: We often use indentation for union cases in the code base.

**Comment by dungpa** [comment]
Looks good. Thanks for doing it.

File: src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs (hunk 2 above)
Comment (smoothdeveloper) [inline src/FSharpVSPowerTools.Core/Symbols/HighlightUsageInFile.fs]: This doesn't seem to work with records, I assume those are seen as (intrinsic) extension members.

**Comment by smoothdeveloper** [comment]
@dungpa thanks for review, I can't fix the record which requires `with`, please let me know if we should do it another way.

**Comment by dungpa** [comment]
I have an example at https://github.com/dungpa/fantomas/blob/dbb019431910a8fbc416d5aa003d79ca0dd6960c/src/Fantomas/FormatConfig.fs#L118-L156.

But it's not important anyway.

**Pull request merged**